docs/architecture/psa-migration/transition-guards.md

```diff
@@ -0,0 +1,290 @@
+This document explains feature guards macros to be used during the transition
+from legacy to PSA in order to determine whether a given cryptographic
+mechanism is available in the current build.
+
+We currently (as of Mbed TLS 3.6) have three sets of feature macros:
+- `PSA_WANT` macros;
+- legacy `MBEDTLS_xxx` macros;
+- transitional `MBEDTLS_xxx` macros that stem from the desire to be able to
+  use crypto mechanisms that are only provided by a driver (G5 in
+`strategy.md`).
+
```

Comment:
Macros defined in `include/mbedtls/config_adjust_legacy_*.h` only, used outside of `include/mbedtls/*.h` and not mentioned here:

* `MBEDTLS_BLOCK_CIPHER_xxx_VIA_xxx` (used in library, test dependencies)
* `MBEDTLS_BLOCK_CIPHER_SOME_PSA` (used in library, test helpers, test code)
* `MBEDTLS_MD_xxx_VIA_PSA` (used in library, test dependencies)
* `MBEDTLS_PSA_UTIL_HAVE_ECDSA` (used in library, test dependencies)

We should document them as well, but that's not a blocker for this pull request.